apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/ILogEntryLabelProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+
+namespace Google.Cloud.Diagnostics.AspNetCore
+{
+    /// <summary>
+    /// Provides a hook to augment labels when a log entry is being logged.
+    /// </summary>
+    public interface ILogEntryLabelProvider
+    {
+        /// <summary>
+        /// Invokes the provider to augment log entry labels.
+        /// </summary>
+        /// <param name="labels">A dictionary of log entry labels.</param>
+        void Invoke(Dictionary<string, string> labels);
```

Comment:
Perhaps it's nice if this would be broader and gives you a chance to augment the whole `LogEntry` instance? The `LogEntry` class seems a bit low-level though, as it's the actual protobuf message being sent over the wire. What do you think @iantalarico?

Reply:
I'm sure someone may come up with a reason for wanting to modify this but if you look at the [definition](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry) I don't see much need to change much else.  

I think sticking with the labels here is good.

Reply:
The only use case I can come up with is if you would want to _inspect_ the payload of the log entry and add some custom labels based on whatever.

I agree that just augmenting the labels is probably fine though.